src/main/java/org/libertybikes/ai/AILogic.java

```diff
@@ -41,118 +41,58 @@
         OPEN,
         FIXED_OBSTACLE,
         MOVING_OBSTACLE,
-        PLAYER
+        PLAYER,
+        PLAYER_TRAIL
     }
 
     private final GAME_SLOT[][] gameBoard = new GAME_SLOT[BOARD_SIZE][BOARD_SIZE];
-    private final String myPlayerId;
-    private boolean filledFixedObstacles = false;
-    private DIRECTION currentDirection = DIRECTION.RIGHT;
-    private int currentX;
-    private int currentY;
 
     public AILogic(String myId) {
-        myPlayerId = myId;
         for (int x = 0; x < BOARD_SIZE; x++)
             Arrays.fill(gameBoard[x], GAME_SLOT.OPEN);
     }
 
     /*
-     * This should contain the main logic of the AI, it receives a JSON straing of all the
+     * This should contain the main logic of the AI, it receives a JSON string of all the
      * objects on the board, players and obstacles.
      */
     public DIRECTION processAiMove(GameTick gameTick) {
-        updateBoard(gameTick);
-
-        System.out.println("Currently facing: " + currentDirection);
-        DIRECTION bestDirection = currentDirection;
-        int bestDirectionSlots = 0;
-        for (DIRECTION d : DIRECTION.values()) {
-            int freeSlots = freeSlots(d);
-            if (freeSlots > bestDirectionSlots) {
-                bestDirection = d;
-                bestDirectionSlots = freeSlots;
-            }
-        }
-        currentDirection = bestDirection;
-        System.out.println("  will face: " + currentDirection);
-        return currentDirection;
-    }
-
-    private int freeSlots(DIRECTION lookingDirection) {
-        int xIncrement = 0, yIncrement = 0;
-        if (lookingDirection == DIRECTION.LEFT) {
-            xIncrement = -1;
-            yIncrement = 0;
-        } else if (lookingDirection == DIRECTION.UP) {
-            xIncrement = 0;
-            yIncrement = -1;
-        } else if (lookingDirection == DIRECTION.RIGHT) {
-            xIncrement = 1;
-            yIncrement = 0;
-        } else { // down
-            xIncrement = 0;
-            yIncrement = 1;
-        }
-
-        int freeSlots = 0;
-        int x = this.currentX + (xIncrement * 2);
-        int y = this.currentY + (yIncrement * 2);
-        // TODO: Should be looking 3 slots wide in each direction
-        // rather than just 1 slot wide in each direction
-        while (x < BOARD_SIZE && y < BOARD_SIZE &&
-               x > 0 && y > 0 &&
-               gameBoard[x][y] == GAME_SLOT.OPEN) {
-            freeSlots++;
-            x += xIncrement;
-            y += yIncrement;
-        }
-        System.out.println("  Found " + freeSlots + " free slots looking " + lookingDirection);
-        return freeSlots;
-    }
-
-    private void updateBoard(GameTick currentBoard) {
```

Comment:
lets leave this method in place. For any parts that don't compile such as `currentBoard.obstacles` lets leave them commented out